pkgs/desktops/gnome/extensions/g3kb-switch/default.nix

```
@@ -0,0 +1,41 @@
+{ lib
+, stdenv
+, cmake
+, pkg-config
+, glib
+, fetchFromGitHub
+  # , bash-completion
+}:
+stdenv.mkDerivation rec {
+  pname = "gnome-shell-extension-g3kb-switch";
+  version = "1.2";
+  src = fetchFromGitHub {
+    owner = "lyokha";
+    repo = "g3kb-switch";
+    rev = version;
+    sha256 = "sha256-Nq2psfy51hZQEm57PUSvs17Pp2z6pe30h9BOLbOYV4k=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+    # bash-completion
+  ];
+  buildInputs = [
+    glib
+  ];
+  prePatch = ''
```

```suggestion
  postPatch = ''
```

otherwise you might get conflicts in patches.